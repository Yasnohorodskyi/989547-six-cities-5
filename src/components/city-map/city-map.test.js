import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";

import CityMap from "./city-map";

const mockStore = configureStore()({
  PROCESS: {
    hoveredOffer: {
      id: 213,
      location: {
        latitude: 234,
        longitude: 999,
        zoom: 12
      }
    }
  }
});

const offers = [
  {
    id: 213,
    location: {
      latitude: 3123124,
      longitude: 425,
      zoom: 12
    }
  },
  {
    id: 123,
    location: {
      latitude: 3123125,
      longitude: 4250.6,
      zoom: 12
    }
  }
];

const MapTypes = {
  SMALL: `SMALL`,
  BIG: `BIG`
};

describe(`Should Map render correctly`, () => {
  it(`Should map of whole city render`, () => {
    const div = document.createElement(`div`);
    div.id = `map`;
    document.body.appendChild(div);
    const tree = renderer
    .create(
        <Provider store={mockStore}>
          <MemoryRouter>
            <CityMap
              offers={offers}
              styles={{}}
              hoveredOffer={{}}
              type={MapTypes.SMALL}
            />
          </MemoryRouter>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
