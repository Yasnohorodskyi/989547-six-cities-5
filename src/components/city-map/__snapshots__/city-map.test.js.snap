// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Map render correctly Should map of whole city render 1`] = `
<div
  id="map"
  style={
    Object {
      "height": "100%",
    }
  }
/>
`;
